### ISSUE TITLE
Polly - Neural voices are not returned for PHP SDK

### ISSUE THREAD

**Issue by yosun** [aws/aws-sdk-php#1992]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/aws-sdk-php/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-php-sdk) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-php/issues) and didn't find any solution
  
**Describe the bug**
When I pass in the neural parameter ( 'Engine' => 'neural' ) for the PHP SDK, the returned voice is not the neural voice but the standard voice. I have tried this for Brian and the other voices listed as neural.  https://docs.aws.amazon.com/polly/latest/dg/ntts-voices-main.html

**Version of AWS SDK for PHP?**
3.91.4

**Version of PHP (`php -v`)?**
5.5.38

**To Reproduce (observed behavior)**
Steps to reproduce the behavior (please share code or minimal repo)

```php
    $arrtemp = [
        'Text' => 'Hello, do I sound neural to you? Compare me on standard. I sound just the same using AWS PHP SDK!',
        'OutputFormat' => 'mp3',
        'VoiceId' => 'Brian',
        'Engine' => 'neural',
        'region' => 'us-east-1'
    ]; 

    $result = $client->synthesizeSpeech($arrtemp);
```

**Expected behavior**

The mp3 should sound different on the neural engine vs standard engine. 

 

**Additional context**

**Comment by diehlaws** [comment]
Hi @yosun, thanks for reaching out to us about this. I've tried reproducing the described behavior on the AWS SDKs for PHP, Go, and Python, as well as using the Polly web console. While the differences in inflection between the standard and neural engines are minimal when using Brian as the `VoiceId`, they do not sound identical. Using Kimberly as the `VoiceId` shows a more obvious difference between standard and neural engines, I have attached a zipped folder with sample audio files for each of these voices using the standard and neural engines, each voice's standard engine output sounds identical regardless of the language used to issue the `synthesizeSpeech` call, likewise with the neural engine's output.

As a result, the `Engine` parameter of this call appears to be working as expected on the AWS SDK for PHP. Please do let us know if you have further questions about this for us.

[pollyVoices.zip](https://github.com/aws/aws-sdk-php/files/4506174/pollyVoices.zip)

**Comment by yosun** [comment]
Can you include the exact parameters of the call you are sending? 

Can you also try applying the conversational tag? It generates an error when called using the PHP SDK. 

https://docs.aws.amazon.com/polly/latest/dg/supportedtags.html#conversational-tag

**Comment by diehlaws** [comment]
@yosun Here's the code I'm using to test the different voices and engines. 

```php
<?php

require_once("../../vendor/autoload.php");

use Aws\Polly\PollyClient;

$client = new PollyClient([
    'region' => 'us-east-1',
    'version' => 'latest',
]);

$engines = ['standard','neural'];

foreach($engines as $engine) {
    $text = "Synthesizing " . $engine . " speech using AWS PHP SDK. Does this sound " . $engine . "?";
    try {
        $arrtemp = [
            'Text' => $text,
            'OutputFormat' => 'mp3',
            'VoiceId' => 'Kimberly',
            'Engine' => $engine,
        ]; 
    
        $result = $client->synthesizeSpeech($arrtemp);
    
    } catch (\Aws\Exception\AwsException $e) {
        echo "Error: {$e->getMessage()}";
    }
    if (!empty($result['AudioStream'])) {
        $audio = $result['AudioStream'];
        file_put_contents('./pollyVoices/Kimberly' . $engine . 'Php.mp3', $audio);
    }
}
```

I'll test changing the `TextType` to `ssml` to apply the conversational tag and will update the issue with my results. What is the error you're receiving when applying this tag to the text used for this call?

**Comment by diehlaws** [comment]
@yosun I was able to receive a successful response with the expected audio output when using `'TextType' => 'ssml'` along with the sample text shown in the documentation page you linked. As noted in that page, only the Matthew and Joanna voices currently support this tag. Following are the parameters I used that resulted in a successful speech synthesis.

```php
$engine = 'neural';

$text = "<speak> 
<amazon:domain name=\"conversational\"> 
I really didn't know how this morning was going to start. And if I had known, I think 
I might have just stayed in bed. Even with a cat sleeping on my head and my dog deciding 
I really didn't need to move my legs and falling asleep across both of them. As it was,
I stayed there as long as my bladder would let me.
</amazon:domain> 
</speak>";

$arrtemp = [
    'Text' => $text,
    'TextType' => 'ssml',
    'OutputFormat' => 'mp3',
    'VoiceId' => 'Matthew',
    'Engine' => $engine,
]; 

$result = $client->synthesizeSpeech($arrtemp);
```

**Comment by yosun** [comment]
So I tried your exact code (and also set version to latest - although afaik there is only the 2016-06-10 version?) and somehow I still get this error: 

` 
Error executing "SynthesizeSpeech" on "https://polly.us-east-1.amazonaws.com/v1/speech"; AWS HTTP error: Client error: `POST https://polly.us-east-1.amazonaws.com/v1/speech` resulted in a `400 Bad Request` response: {"message":"This voice does not support one of the used SSML features"} ValidationException (client): This voice does not support one of the used SSML features - {"message":"This voice does not support one of the used SSML features"}`

**Comment by yosun** [comment]
`Stack trace: #0 /home/vendor/guzzlehttp/guzzle/src/Middleware.php(66): GuzzleHttp\Exception\RequestException::create(Object(Gu in /home/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 191`

**Comment by yosun** [comment]
Am I just using the wrong version? Everything works when I set the engine to standard and remove the conversation tags... 

 Which version of the SDK are you using - and also which version of the polly API are you using?

**Comment by diehlaws** [comment]
@yosun My apologies, it seems like the discrepancy in behavior here comes down to the different versions of the SDK being used. After downgrading my version of the SDK from 3.134.2 to 3.91.4 I'm getting the same validation exception (`{"message":"This voice does not support one of the used SSML features"}`) when executing the above snippet. To address your question, the Polly API currently has only one version published (`2016-06-10`) but using `'version' => 'latest'` is a more convenient way to use the most recently published version of a service's API for testing. Given that there is only one published version of the service's API using either `version` string in the service client's config should be functionally identical in this case.

Looking at the SDK's [changelog](https://github.com/aws/aws-sdk-php/blob/master/CHANGELOG.md#31082---2019-07-30) I see that support for Polly's neural engine was added in version 3.108.2, which would explain why you're not hearing any difference in the audio when comparing against the standard engine. Upgrading the SDK from 3.91.4 to 3.108.2 resulted in the most recent snippet I posted working as expected. If you want to use the neural engine for Polly via the AWS SDK for PHP you'll have to upgrade to at least that version of the SDK.
